### PR TITLE
Skill - Technician

### PR DESCRIPTION
**Pull request by JesterWizard**
If the user is using an E-ranked weapon, boost damage by 50% of the weapon might.

### CODE REVIEW

**Comment by MokhaLeee** [comment]
reviewed with comment

File: Wizardry/Core/BattleSys/Source/PreBattleCalc.c [hunk 1]
```diff
@@ -20,10 +20,10 @@ void ComputeBattleUnitAttack(struct BattleUnit * attacker, struct BattleUnit * d
     {
         status = status * 2;
 
-#if (defined(SID_Resourceful) && (SID_Resourceful < MAX_SKILL_NUM))
-        if (SkillTester(&attacker->unit, SID_Resourceful))
-            status = status * 2;
-#endif
+        #if (defined(SID_Resourceful) && (SID_Resourceful < MAX_SKILL_NUM))
```
[the remark this answers is not in the document]
Comment (JesterWizard) [reply Wizardry/Core/BattleSys/Source/PreBattleCalc.c]: Reverted.

File: Wizardry/Core/BattleSys/Source/PreBattleCalc.c [hunk 2]
```diff
@@ -456,6 +456,18 @@ STATIC_DECLAR void PreBattlePostCalcSkills(struct BattleUnit * attacker, struct
...
     }
 #endif
+
+#if (defined(SID_Technician) && (SID_Technician < MAX_SKILL_NUM))
+    if (SkillTester(unit, SID_Technician))
+    {
+        if (GetItemRequiredExp(attacker->weapon) < 31)
+        {
+            int itemMight = GetItemMight(attacker->weapon);
+            attacker->battleAttack += itemMight/2;
```
[the remark this answers is not in the document]
Comment (JesterWizard) [reply Wizardry/Core/BattleSys/Source/PreBattleCalc.c]: Done.

File: Wizardry/Core/BattleSys/Source/PreBattleCalc.c [hunk 3]
```diff
@@ -456,6 +456,18 @@ STATIC_DECLAR void PreBattlePostCalcSkills(struct BattleUnit * attacker, struct
         attacker->battleAvoidRate += (15 - (turnNumber-1));
     }
 #endif
+
+#if (defined(SID_Technician) && (SID_Technician < MAX_SKILL_NUM))
+    if (SkillTester(unit, SID_Technician))
+    {
+        if (GetItemRequiredExp(attacker->weapon) < 31)
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/PreBattleCalc.c:463]: use macro `WPN_EXP_D`

Comment (JesterWizard) [reply Wizardry/Core/BattleSys/Source/PreBattleCalc.c]: Done.

**Pull request merged**